src/providers/cryptoauthlib/hash.rs

```diff
@@ -33,4 +34,35 @@ impl Provider {
             }
         }
     }
+
+    pub(super) fn psa_hash_compare_internal(
+        &self,
+        op: psa_hash_compare::Operation,
+    ) -> Result<psa_hash_compare::Result> {
+        let alg_len = op.alg.hash_length();
+        // calculate input hash
+        let op_compute = psa_hash_compute::Operation {
+            alg: op.alg,
+            input: op.input,
+        };
+        match self.psa_hash_compute_internal(op_compute) {
+            Ok(psa_hash_compute::Result { hash }) => {
+                // compare hashes length
+                if op.hash.len() != alg_len || hash.len() != alg_len {
```

Comment:
https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_hash_compare.html

> `PsaErrorInvalidArgument`: The length of `input` or `hash` does not match the hash size for `alg`.

I was trying to cover length of `input` part in this statement. I believe it meant to be length of calculated hash from input because input message can be any size. So I think according to this page type of logged error is correct.

Reply:
The `input` part is actually about the input size - there are limits on it, usually something on the order of 2^64 (so it's more of a theoretical limit than a practical one) so I wouldn't bother checking that - the machine would be long dead by the time it's able to transfer that much data. (I did check with one of the engineers working on PSA Crypto for this one!)

Reply:
(ps: I'm happy to leave this as is, there's barely any impact from this, it's more of a nit - it's more the logging change that's relevant)